### ISSUE TITLE
graph dependencies

### ISSUE THREAD

**Issue by ordirules** [mrocklin/streamz#44]
Right now, graph

```python
from streams import Stream
s1 = Stream()
s2 = Stream()

sout1 = s1.map(lambda x : x +1)
sout1.map(print)

# connect stream 2 to stream 1
s2.map(s1.emit)

from streams.graph import visualize 
visualize(s2, 'stream2.png')
```

Result is this:
http://imgur.com/a/0Z0JA

What would solve this is that rather than using `emit`, we could define a new function function, say `s2.inject(s1)` which would add `s2` as a child to `s1`. The only issue I see here is we'll need an `update` defined for the base stream object, which will just call `emit`. What do you think?
@mrocklin @CJ-Wright 
I'm interested to hear of a better way, thanks!

**Comment by ordirules** [comment]
I'm also happy to do the grunt work on a PR but just want consensus first.

**Comment by CJ-Wright** [comment]
Closed by #45
